modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -488,12 +490,26 @@ const FrontendDataSet = ({
 					/>
 				) : (
 					<ClayEmptyState
-						description={Liferay.Language.get(
-							'sorry,-no-results-were-found'
+						description={
+							emptyState.description ??
+							Liferay.Language.get('sorry,-no-results-were-found')
+						}
+						imgSrc={
+							themeDisplay.getPathThemeImages() +
+							(emptyState.image ?? '/states/search_state.gif')
+						}
+						title={
+							emptyState.title ??
+							Liferay.Language.get('no-results-found')
+						}
+					>
+						{emptyState.actionItems && (
+							<CreationMenu
+								{...emptyState.actionItems}
+								defaultButton
+							/>
```

Comment:
OK, this makes sense, reusing the `CreationMenu` component :+1:  In this case:
1. We don't need to go back and forth with `actionItems`. We can just call the prop `creationMenu`, so it matches how it's used internally.
2. `defaultButton` is a bit vague. Let's call this prop `inEmptyState`

Reply:
I'm fine with that decision in this PR, we can ellaborate a bit in a followup pull if we find the case where the creation menu must not be fully represented in the empty state button.

Reply:
Yes.

If requirements are not clear, it's always better to add a more restrictive API first, and extend it later on if needed. For our cases in data set manager, it is not needed.

Even if we do decide to have more open API, with an additional prop for a separate set of actions in empty state, copying `creationMenu` will be a good fallback, in the case that new prop is not set.

Reply:
> Yes.
> 
> If requirements are not clear, it's always better to add a more restrictive API first, and extend it later on if needed. For our cases in data set manager, it is not needed.
> 
> Even if we do decide to have more open API, with an additional prop for a separate set of actions in empty state, copying `creationMenu` will be a good fallback, in the case the new prop is not set.

Roger that! Just to be sure, did you mean this https://github.com/liferay-frontend/liferay-portal/pull/3399/commits/b5fbf0940044caa098cccdc293bdaa59bad72581 ?

Reply:
To make API more abstract and reusable, we could add something like `triggerProps` and `containerProps`. In them, we can set empty state specifics, like the `d-inline-flex` class. Defaults would need to match existing case.

Reply:
> Just to be sure, did you mean this https://github.com/liferay-frontend/liferay-portal/commit/b5fbf0940044caa098cccdc293bdaa59bad72581 ?

Yep.

Reply:
> To make API more abstract and reusable, we could add something like `triggerProps` and `containerProps`. In them, we can set empty state specifics, like the `d-inline-flex` class. Defaults would need to match standard creation menu cases.
> 
> It might be a bit of over-engineering, but I'd be fine with either approach.

Since this looks a bit more complex, I'm going to steal your sentence,

> it's always better to add a more restrictive API first, and extend it later on if needed

and change it to `inEmptyState` for now!

Reply:
https://github.com/liferay-frontend/liferay-portal/pull/3399/commits/81d3d6cfb91ff1dc52155dadf9c3565cfcbf7e42